docs/core/diagnostics/metrics-collection.md

```diff
@@ -0,0 +1,404 @@
+    Gen 2 Size (B)                                           203,248
+    LOH Size (B)                                             933,216
+    Monitor Lock Contention Count (Count / 1 sec)                  0
+    Number of Active Timers                                        1
+    Number of Assemblies Loaded                                   39
+    ThreadPool Completed Work Item Count (Count / 1 sec)           0
+    ThreadPool Queue Length                                        0
+    ThreadPool Thread Count                                        3
+    Working Set (MB)                                              30
+```
+
+See the [dotnet-counters](dotnet-counters.md) for more information about the tool or [builtin metrics](available-counters.md) to learn
+more about metrics that are available out of the box in .NET.
+
+### Viewing metrics in Grafana with OpenTelemetry and Prometheus
+
+#### Prerequisites
+
+- [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet) or a later version
+
+#### Overview
+
+[OpenTelemetry](https://opentelemetry.io/) is a vendor-neutral open-source project supported by the
+[Cloud Native Computing Foundation](https://www.cncf.io/) that aims to standardize generating and collecting telemetry for
+cloud-native software. The .NET team has designed our built-in platform metric APIs to be compatible with this
+standard to make integration straightforward for any .NET developers that wish to use it. At the time of writing support for
+OpenTelemetry metrics is relatively new but [Azure Monitor](https://docs.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-overview)
+and many major APM vendors have endorsed it and have integration plans underway.
+
+This example shows one of the integrations available now for OpenTelemetry metrics using the very popular OSS 
+[Prometheus](https://prometheus.io/) and [Grafana](https://grafana.com/) projects. The metrics data will flow like this:
+
+1. The .NET metric APIs collect measurements from our sample application
+2. The OpenTelemetry library running inside the same process aggregates these measurements
+3. The Prometheus exporter library makes the aggregated data available via an HTTP
+metrics endpoint. 'Exporter' is what OpenTelemetry calls the libraries that transmit
+telemetry to vendor-specific services.
+4. A Prometheus service, potentially running on a different machine, polls the
+metrics endpoint, reads the data and stores it in a database for long-term persistence.
+5. The Grafana service, potentially running on a different machine, queries the data
+stored in Prometheus and displays it to engineers on a web-based monitoring dashboard.
+
+#### Configure the sample application to use OpenTelemetry's Prometheus exporter
+
+1. Add a reference to the OpenTelemetry Prometheus exporter to the sample application:
+
+```dotnetcli
+dotnet add package OpenTelemetry.Exporter.Prometheus --version 1.2.0-beta1
```

Comment:
>--version 1.2.0-beta1 [](http://example.com/codeflow?start=53&length=21)

I am not sure if it is worth it to mention can use `--prerelease` instead if using .NET 5.0 SDK?

Reply:
Given that the OpenTelemetry crew might make breaking changes between pre-release builds it is probably safer to pin the exact version rather than allowing it to float via --prerelease. Once an RTM version is available I'd love to remove the '--version' from the docs entirely.

I did go ahead and add a warning that this is pre-release functionality, that breaking changes are possible, and that we only tested with this exact version.